web/src/stream_popover.ts

```diff
@@ -741,7 +742,11 @@ export async function build_move_topic_to_stream_popover(
             return false;
         }
         let {new_topic_name} = get_params_from_form();
-        if (!settings_data.user_can_move_messages_to_another_topic()) {
+        if (
+            !settings_data.user_can_move_messages_to_another_topic() &&
+            (stream_widget_value !== current_stream_id ||
+                !stream_data.user_can_move_messages_within_channel(current_stream_id))
```

Comment:
I don't understand why this hunk is correct.

Reply:
Oh is this the same as https://github.com/zulip/zulip/pull/34867/commits/c09edc98b2ebd86d41e3e5912f54de63dea32ad8#diff-26aa930b41caa2a7e31ee5622de9b7629fbe56f84c27a87315fa8189420d45e2R1488? Probably worth a similar comment.